drivers/rtc/rtc_max31329.c

```diff
@@ -0,0 +1,570 @@
+/*
+ * Copyright (c) 2024-2025 Wimansha Wijekoon <Wimanshahb@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Following is the Driver for max31329 RTC from analog device
+ * rtc_pcf8563.c has been used as a reference to develop the base code
+ *
+ * with this driver following features been supported
+ *
+ * - RTC Alarm 1 interrupt
+ * - Trickle charging
+ * - Event input (uses same GPIO as alarm interrupt)
+ *
+ *Example of DT segment
+ *
+ *    rtc3: max31329@68 {
+ *        compatible = "adi,max31329";
+ *        reg = <0x68>;
+ *        status = "okay";
+ *        pvft = <0x1>;
+ *        trickle = <0x1>;
+ *        rtc-inta-gpios = <&gpio0 24 GPIO_ACTIVE_LOW>;
+ *        digital-in-enable;
```

Comment:
Example is missing the property `alarms-count = <2>;` which is inherited from `rtc-device.yaml`. Its used by apps at compile time to know how many alarms are present.